### DISCUSSION TITLE
Question: is insert operation thread safe?

### DISCUSSION THREAD

**Discussion by VPaulV** [MrPigss/BetterJSONStorage#7] [Q&A] [answered]
Hi Guys,

Sorry for the stupid question, could not find this in README. Is insert operation thread safe? I guess I can do simultaneous reads without any problem, but what about writes?

**Comment by ankraft** [comment, marked as answer]
Unfortunately, TinyDB in general is not thread-safe and you might need to use locking. See https://github.com/msiemens/tinydb/discussions/452

**Comment by VPaulV** [comment]
I see, thank you! I guess I can use something like tinyrecord for this purpose

**Comment by ankraft** [comment]
I am not sure that tinyrecord provides thread safety. It provides transactional requests.
In my experience with TinyDB it might be necessary to lock **all** access for each data base, not only each table.
It is a bit more overhead but actually not that complicated:

``` python
from threading import Lock
myLock = Lock()
...
with myLock:
    tinydbTable.operation()
```
